PBN RCT.Rmd

```diff
@@ -27,157 +27,210 @@ if(!require(openxlsx)){install.packages('openxlsx')}
   mutate(treatment_group = Condition == "Intervention")
 
-#Select the participants being included in analysis (per data integrity checks, should be N=220)
-included_participants <- merged_data %>%
-  filter(Participant_ID %in% c(40077, 15186, 11697, 95674, 40889, 84608, 61437, 16432, 22485, 24378, 21140, 58776, 13649, 83679, 62516, 52326, 20457, 39455, 64886, 35756, 65322, 86883, 72424, 92104, 33504, 38713, 96986, 28301, 14617, 14806, 93863, 13842, 31849, 74139, 53775, 57712, 96120, 49399, 45841, 40397, 75164, 71675, 30277, 58650, 18169, 41917, 50884, 70371, 48434, 18974, 18856, 17231, 56049, 68213, 93764, 15333, 98726, 80569, 63962, 66481, 40376, 83329, 90314, 59978, 66920, 35067, 15025, 77121, 89229, 82287, 96711, 11308, 53611, 57517, 47321, 43097, 74462, 11290, 49908, 24314, 69311, 70500, 49864, 69898, 74951, 38601, 60126, 26551, 58830, 21592, 17149, 31067, 39743, 71511, 45789, 84417, 56534, 98233, 84267, 68200, 76371, 45565, 48031, 19978, 25987, 38488, 25643, 86804, 86300, 18348, 90947, 53863, 17032, 60501, 27455, 62781, 88241, 55765, 39606, 66912, 67236, 11868, 68267, 35553, 95389, 80941, 15777, 53969, 95004, 43258, 64019, 99949, 72626, 47331, 50401, 96007, 65187, 31188, 93980, 76870, 32309, 85411, 85344, 48970, 54988, 49034, 77719, 78040, 86763, 33280, 23158, 87771, 21842, 19199, 96320, 64719, 78881, 49236, 26869, 65509, 38538, 73614, 80888, 55845, 86114, 22584, 81514, 92289, 86314, 65863, 77450, 25638, 88046, 99374, 13638, 35047, 38572, 26776, 86023, 43295, 20973, 53220, 29614, 62193, 42215, 41799, 93664, 50466, 78161, 85350, 87127, 15065, 11827, 63743, 20364, 17670, 44268, 48593, 96225, 71632, 97410, 90795, 49102, 68198, 16855, 64531, 13302, 57955, 21313, 13023, 69913, 75263, 93022, 48022, 22684, 17052, 72038, 56590, 56815, 76432))
+#Select variables that will be used for analysis / in imputation
+###SINCE REMOVING TIMING AND TEXT-BASED VARIABLES, WILL NEED TO LOOK AT SEPERATELY ELSEWHERE###
+###WHAT ELSE SHOULD WE INCLUDE####
+included_participants <- included_participants %>%
+   select(Participant_ID, treatment_group, starts_with(c("b_dem", "b_current", "b_past", "b_unmet", "b_hunger", "b_bhs", "b_shs_agency", "b_biss", "b_fas", "b_biaaq", "b_chime", "b_mfq", "b_edeq", "b_mddi", "pi_pfs", "pi_bhs", "pi_shs_agency", "pi_biss", "pi_fas", "pi_biaaq", "pi_chime", "f_new", "f_bhs", "f_shs_agency", "f_biss", "f_fas", "f_biaaq", "f_chime", "f_mfq", "f_edeq", "f_mddi"))) %>%
+  select(-contains("free"), -contains("TEXT"), -contains("_tim_"), -contains("pfs_like"), -contains("pfs_change"), -contains("pfs_other")) %>%
+  mutate_all(as.numeric)
+###NEED TO CHECK### - mutate_all(as.numeric) produces two warning messages that introduce NAs in b_dem_race_ethnicity and b_dem_gender but I'm not sure why. I am pretty sure missForest() can only work with numeric and categorical variables (meaning I think we have to override variables like treatment_group that were originally logical). Will this cause problems later?
```

Comment:
Should these variables be converted to factor types instead?

What comes up when you type `as.numeric(included_participants$b_dem_gender)` into the console?